File: CS/XtraReport2.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace Rep3
{
    public partial class XtraReport2 : DevExpress.XtraReports.UI.XtraReport
    {
        public XtraReport2()
        {
            InitializeComponent();
        }

        private void xrSubreport1_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e)
        {
            // Create the detail report 
            SubReport reportProducts = xrSubreport1.ReportSource as SubReport;

            // Pass the CategoryID parameter from the master report.
            reportProducts.CategoryID = (int)this.GetCurrentColumnValue("CategoryID");

            if (reportProducts.CategoryID % 2 == 1)
                e.Cancel = true;

        }

    }
}

File: VB/XtraReport2.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace Rep3
	Partial Public Class XtraReport2
		Inherits DevExpress.XtraReports.UI.XtraReport
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub xrSubreport1_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles xrSubreport1.BeforePrint
			' Create the detail report 
			Dim reportProducts As SubReport = TryCast(xrSubreport1.ReportSource, SubReport)

			' Pass the CategoryID parameter from the master report.
			reportProducts.CategoryID = CInt(Fix(Me.GetCurrentColumnValue("CategoryID")))

			If reportProducts.CategoryID Mod 2 = 1 Then
				e.Cancel = True
			End If

		End Sub

	End Class
End Namespace
